### PR TITLE
Restart NBLS when an extension providing some NB cluster is installed/uninstalled.

### PR DESCRIPTION
**Pull request by dbalek**
Language Server should be restarted whenever an extension providing its part (e.g. nbjavac) is installed/uninstalled.

### CODE REVIEW

File: java/java.lsp.server/vscode/src/extension.ts [hunk 1]
```diff
@@ -66,7 +67,7 @@ export function enableConsoleLog() {
 }
 
 export function findClusters(myPath : string): string[] {
```
Comment (JaroslavTulach) [inline java/java.lsp.server/vscode/src/extension.ts:69]: This is an exported function, I'd prefer to keep it pure (e.g. without side effects).

File: java/java.lsp.server/vscode/src/extension.ts [hunk 2]
```diff
@@ -156,6 +157,13 @@ export function activate(context: ExtensionContext): VSNetBeansAPI {
 
     // find acceptable JDK and launch the Java part
     findJDK((specifiedJDK) => {
+        context.subscriptions.push(vscode.extensions.onDidChange(() => {
+            const currentClustersSorted = currentClusters.slice().sort();
```
Comment (JaroslavTulach) [inline java/java.lsp.server/vscode/src/extension.ts:161]: keep the global state somewhere here, if possible.

File: java/java.lsp.server/vscode/src/extension.ts [hunk 3]
```diff
@@ -156,6 +157,13 @@ export function activate(context: ExtensionContext): VSNetBeansAPI {
...
+            const newClusters = findClusters(context.extensionPath);
+            if (newClusters.length !== currentClustersSorted.length || newClusters.sort().find((value, index) => value !== currentClustersSorted[index])) {
```
Comment (JaroslavTulach) [inline java/java.lsp.server/vscode/src/extension.ts:163]: Wow. That's a functional programming!

**Review by JaroslavTulach — APPROVED** [review]
Please keep the `findClusters` function without side effects. I'd be nice to expand the [findClusters test](https://github.com/apache/netbeans/blob/983cae8922d52e11657661fab95a5420349df10a/java/java.lsp.server/vscode/src/test/suite/extension.test.ts#L35). Otherwise OK.

File: java/java.lsp.server/vscode/src/extension.ts (hunk 1 above)
Comment (dbalek) [reply java/java.lsp.server/vscode/src/extension.ts]: Fixed.

File: java/java.lsp.server/vscode/src/extension.ts (hunk 2 above)
Comment (dbalek) [reply java/java.lsp.server/vscode/src/extension.ts]: Fixed.

**Pull request merged**